### ISSUE TITLE
adding X_FORWARDED_PREFIX for wsgi compatible apps

### ISSUE THREAD

**Issue by gbrault** [jupyterhub/jupyter-server-proxy#216]
In file jupyter-server-proxy/jupyter_server_proxy/handlers.py [in line ](https://github.com/jupyterhub/jupyter-server-proxy/blob/ee403c608f9b92ed712576e50be4c4539ca939a7/jupyter_server_proxy/handlers.py#L173) adding the following code

```
headers['X_FORWARDED_PREFIX'] = context_path
```

This allows for example flask/werkzeug app to be able to run seamless in jupyter lab context

adding the following lines in the app does the job from flask perspective:

```
from  werkzeug.middleware.proxy_fix import ProxyFix
....
After creating the Flask app:
app.wsgi_app = ProxyFix(app.wsgi_app, x_for=1, x_host=1, x_prefix=1)
```

With this, jypyter server proxy makes flask/werkzeug apps working seamlesly! 

And making sure the front proxy is passing the appropriate headers, with nginx for example
x-for, x-host
```
real_ip_recursive on;

server {
  client_max_body_size 200M;
  server_name jod.seadev.xyz;
  
  add_header X-Forwarded-For $proxy_add_x_forwarded_for;
  add_header X-Remote-IP $remote_addr;
  add_header X-Forwarded-Host jod.seadev.xyz;

  location / {
    proxy_set_header        Host $host;
    proxy_set_header        X-Forwarded-Host jod.seadev.xyz;
    proxy_set_header        X-Real-IP $remote_addr;
    proxy_set_header        X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header        X-Forwarded-Proto $scheme;
    proxy_pass              http://127.0.0.1:8888;
    # WebSocket support (nginx 1.4)
    proxy_http_version 1.1;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection "upgrade";
    proxy_read_timeout  36000s;    
  }
}
```

I can send a pull request if you agree?

**Comment by yuvipanda** [comment]
Thank you for digging into this! Would love a pull request, this would indeed be very useful!

**Comment by ryanlovett** [comment]
Fixed by #217.
